pennylane/devices/qubit/preprocess.py

```diff
@@ -258,15 +259,18 @@ def batch_transform(
 
     Args:
         circuit (.QuantumTape): the circuit to preprocess
+        execution_config (.ExecutionConfig): execution configuration with configurable
+            options for the execution.
 
     Returns:
         tuple[Sequence[.QuantumTape], callable]: Returns a tuple containing
         the sequence of circuits to be executed, and a post-processing function
         to be applied to the list of evaluated circuit results.
     """
-    # Check whether the circuit was broadcasted
-    if circuit.batch_size is None:
-        # If the circuit wasn't broadcasted, no action required
+    # Check whether the circuit was broadcasted or if the diff method is anything other than adjoint
+    if circuit.batch_size is None or execution_config.gradient_method != "adjoint":
+        # If the circuit wasn't broadcasted, or if built-in PennyLane broadcasting
+        # can be used, then no action required
```

Comment:
Hey @eddddddy! I have a random fly-by question. Are these preprocessing functions general ones, meant to be used for any qubit device, or are they implementation details of `default.qubit2`, and not meant to be used for other devices?

Reply:
As far as I understand, these are only to be used by `default.qubit2`.

Reply:
Now that the new transforms stuff is starting to get merged, we are going to make ransforms for all these steps make it easy to pick and choose what steps a particular device wants.